File: BBIS.Api/ClientApp/src/models/UserAccounts/UpdateUserAccountDto.ts
import { IUserAccountBaseDto } from './IUserAccountBaseDto';

export interface IUpdateUserAccountDto extends IUserAccountBaseDto {
    UserAccountId: Guid | null;
    PasswordHasChange: boolean;
    UpdatedPassword: string | null;
    RoleIds: Array<Guid>;
    IsActive: boolean;
    UserModuleIds: Array<Guid>;
}

export class UpdateUserAccountDto implements IUpdateUserAccountDto {
    UserAccountId = null;
    PasswordHasChange = false;
    UpdatedPassword =  null;
    RoleIds = [];
    IsActive = false;
    Username = null;
    Firstname = null;
    Lastname = null;
    EmailAddress = null;
    EventRole = null;
    UserModuleIds = [];
}

File: BBIS.Application/DTOs/UserAccount/UpdateUserAccountDto.cs
namespace BBIS.Application.DTOs.UserAccount
{
    public class UpdateUserAccountDto: UserAccountBaseDto
    {      
        public Guid UserAccountId { get; set; }
        public bool PasswordHasChange { get; set; }
        public string? UpdatedPassword { get; set; }
        public bool IsActive { get; set; }
        public List<Guid> RoleIds { get; set; }
        public List<Guid> UserModuleIds { get; set; }
    }
}
